datalad/support/gitrepo.py

```diff
@@ -988,7 +988,16 @@ def __eq__(self, obj):
     @classmethod
     def is_valid_repo(cls, path):
         """Returns if a given path points to a git repository"""
-        return (Path(path) / '.git').exists()
+        path = Path(path) / '.git'
+        # the aim here is to have this test as cheap as possible, because
+        # it is performed a lot
+        # recognize two things as good-enough indicators of a present
+        # repo: 1) a non-empty .git directory (#3473) and 2) a pointer
+        # file or symlink
+        return path.exists() and (
+            (path.is_dir() and any(path.iterdir())) \
+            or not path.is_dir()
```

Comment:
Very unlikely to be noticeable in practice, but since the comment says the aim is "as cheap as possible", you could avoid a is_dir() call:

```diff
diff --git a/datalad/support/gitrepo.py b/datalad/support/gitrepo.py
index c6658d010..8dbadd041 100644
--- a/datalad/support/gitrepo.py
+++ b/datalad/support/gitrepo.py
@@ -995,9 +995,8 @@ def is_valid_repo(cls, path):
         # repo: 1) a non-empty .git directory (#3473) and 2) a pointer
         # file or symlink
         return path.exists() and (
-            (path.is_dir() and any(path.iterdir())) \
-            or not path.is_dir()
-        )
+            not path.is_dir() or
+            any(path.iterdir()))
 
     @staticmethod
     def get_git_dir(repo):
```

Reply:
That is better. Thx!

Reply:
Done.